server.js

```diff
@@ -9,11 +11,102 @@ app.use(cors());
 //Use this array as your (in-memory) data store.
 const bookings = require("./bookings.json");
 
-app.get("/", function (request, response) {
-  response.send("Hotel booking server.  Ask for /bookings, etc.");
+//Create a new booking
+app.post("/bookings", (req, res) => {
+  const {
+    title,
+    firstName,
+    surname,
+    email,
+    roomId,
+    checkInDate,
+    checkOutDate,
+  } = req.body;
+
+  //check all field been filled up
+  if (
+    !title ||
+    !firstName ||
+    !surname ||
+    !email ||
+    !roomId ||
+    !checkInDate ||
+    !checkOutDate
+  ) {
+    res.json({ msg: "newBooking has empty field" });
+  } //validate email
+  else if (!emailValidator.validate(email)) {
+    res.status(400).json({ msg: "please enter the valid email" });
+  } else if (!moment(checkOutDate).isAfter(checkInDate)) {
+    res.status(400).json({ msg: "checkOutDate is after checkInDate" });
+  } else {
+    const newBooking = req.body;
+    //creating id for the new booking
+
+    newBooking.id = Math.max(...bookings.map((booking) => booking.id), 0) + 1;
```

Comment:
Looks like the `id` is just it's an incremental number, could booking ID be

```
newBooking.id = bookings.length + 1; 
```

Reply:
I didn't use array.length to generate booking IDs because I thought deleted bookings might cause non-unique IDs.